src/lib.rs

```diff
@@ -140,10 +140,7 @@ impl<'a, T> Uri<'a, bitcoin::address::NetworkUnchecked, T> {
         if self.address.is_valid_for_network(network) {
             Ok(self.assume_checked())
         } else {
-            Err(InvalidNetworkError {
-                required: network,
-                found: *self.address.network(),
-            })
+            Err(InvalidNetworkError { required: network })
```

Comment:
Perhaps we can get rid of this whole impl block now that we have `assume_checked` and `require_network` in `rust-bitcoin` now.